src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11539,12 +11603,12 @@ bool isParenVarCommaSyntax()
                         {
                             return true;
                         }
+
+                        var token3 = this.PeekToken(3);
                         // ( x!! , [...]
                         // https://github.com/dotnet/roslyn/issues/58335: https://github.com/dotnet/roslyn/pull/46520#discussion_r466650228
                         if (after.Kind == SyntaxKind.ExclamationToken
-                            && this.PeekToken(3).Kind == SyntaxKind.ExclamationToken
-                            && after.GetTrailingTriviaWidth() == 0
-                            && this.PeekToken(3).GetLeadingTriviaWidth() == 0
+                            && token3.Kind == SyntaxKind.ExclamationToken
```

Comment:
intentionally removed this restriction.

Reply:
correct!  it also means there's uniform 'token handling' that goes in before calling the helper.  none of these scanners (or hte helper) care about having this requirement.  it means they're all resilient to accidental spaces happening as you edit.